### DISCUSSION TITLE
add pagination controls to a repeater container

### DISCUSSION THREAD

**Discussion by gvlx** [Budibase/budibase#3280] [Help]
Hi!

I'm still very new to all the features of Budibase so I was wandering if someone could help me here?

I am using a repeater element in order to have more control over the appearance of the elements but I have a very long list of records. Pagination seems to be in order but I think pagination control is only available for table elements?

Or is there a way to paginate a repeater element, so I don't have to make it extra long?

Thanks for any advice.

Cheers!

**Comment by mjashanks** [comment]
Hi Gerardo.

Pagination is actually part of the `Data Provider` component. It should apply to the repeaters also. You're repeater should be inside a Data Provider (as is a table).

![image](https://user-images.githubusercontent.com/3524181/140735809-69c922a8-19d9-4645-947d-784726e59cf4.png)

**Reply by gvlx** [reply]
It is working for a single data provider designer but once you increase the complexity, it no longer works.

Actually it is a on data-repeater container of a child element:

- data_parent  has limit 1 and no pagination
- data_child ( on relationship field),  limit 3 and paginate

![imagem](https://user-images.githubusercontent.com/67311/140823958-63626a0c-e04d-47f1-91b8-c7eca0d5fa6d.png)

rpt_parent will have pagination controls according to data_parent settings
rpt_child will not have pagination controls, regardless of data_child settings


Maybe there is no pagination for nested repeaters?
Or this should be an error?

**Reply by mjashanks** [reply]
Hmm - I just tested this and it's working ok for me - 

![image](https://user-images.githubusercontent.com/3524181/141082544-b0679e81-1d36-40b8-94ef-17a4ebc81e11.png)

I have the setup
- `One Project has many tasks` 
- my first project has 4 tasks
- I'm setting `Limit 2` on my `Project Data Provider`, with no pagination
- I'm setting `Limit 3` on my `Tasksata Provider`, with pagination enabled

Displays as expected. Have another look at your settings - if you can't figure it out, you can export your app and attach the file in here, or email to support@budibase.com. I can import and look at it!

**Reply by gvlx** [reply]
In my test I have three data levels:
![imagem](https://user-images.githubusercontent.com/67311/144041007-bb2f434a-b5e3-44dc-81a9-0714529505c1.png)

Pagination only appears on the first two levels:
![imagem](https://user-images.githubusercontent.com/67311/144041072-b1ab31f0-b82e-4901-9563-f1ee59f37848.png)

**Reply by gvlx** [reply]
Filed bug #3669
